### ISSUE TITLE
Resume Position Not Working

### ISSUE THREAD

**Issue by Play-AV** [rrirower/DP2SceneGraphWizard#120]
**Describe the bug**
Newest version (1.0.0.5) fixes the hangs / crashes described in my previous issue. However, I'm having some difficulty with resuming videos from both "Continue" and the episode listings themselves. The information is being stored in the Bookmarks correctly, and in *some* but not all cases, correctly displayed within the episode listing, however attempting to resume a video does not actually go to the resume point, it starts at the beginning. 
No errors are being logged and debug logging is on. Will try and dig in later and see if I can find at what point it's going wrong or not being passed so I can maybe provide you some more information. There's also a slight inconsistency with Continue, occasionally after the 'first' item is added, a channel restart is needed for subsequent items to be visible. This is not a major issue I guess. 

Is this an issue you've observed in the latest version?

**To Reproduce**
1. Play content 
2. If content resume position was saved, try to resume the content
3. Content will play from the beginning. 

**Expected behavior**
Content should resume where saved. 

Manifest if that helps. 
```
title=XXX
major_version=40
minor_version=1
build_version=1
FEED_URL=XXX
splash_color=#000000
BACKGROUND_COLOR=#000000
ui_resolutions=hd
USE_RAF=0
OFFER_SUBSCRIPTIONS=0
ENABLE_ROKU_VOICE=0
FAST_FORWARD=0
REWIND_BACK=0
supports_voice_roinput=0
supports_etc_next=0
supports_etc_seek=0
supports_input_launch=1
mm_icon_focus_hd=pkg:/images/channel.jpg
splash_screen_hd=pkg:/images/splash_HD.jpg
splash_screen_fhd=pkg:/images/splash_FHD.jpg


USE_THUMBNAIL_BACKGROUND=false
HIDE_GRID_THUMBNAIL=false
OVERHANG_CLOCK=false
OVERHANG_OPTIONS=false
CONTINUE_WATCHING=true
WATCHED_INDICATOR=true
CONTENT_VOTING=false
ALLOW_CHANNEL_SEARCH=false
SHOW_DETAILS_THUMBNAIL=false
DEBUG_PRINT=1 ```

**Comment by rrirower** [comment]
Can you provide the steps needed to recreate your issue?

**Comment by Play-AV** [comment]
> Can you provide the steps needed to recreate your issue?

It's in the first post

1.Play content
2.If content resume position was saved, try to resume the content
3.Content will play from the beginning.

**Comment by Play-AV** [comment]
Is the Debug zip on here any different than the 1.0.0.5 release?

**Comment by rrirower** [comment]
No

**Comment by rrirower** [comment]
Is the issue related to episodes only?

**Comment by Play-AV** [comment]
> Is the issue related to episodes only?
Seems to not affect "movie" types in the same manner, having said that on one attempt, the movie didn't get added to the "continue row"  nor had any "resume progress" visible. On exiting the channel fully, coming back and playing the same movie (which began from the beginning) navigating a while in, and then going 'back' , the resume progress was saved and clicking resume did indeed resume at the correct position. In the "continue row" clicking that same movie which had resume progress, did also resume from the correct position. The episodes still do not.

**Comment by Play-AV** [comment]
To sum up

Sometimes no progress is saved, or at least visible within the channel. Yes I am skipping in a while and letting it play. 

Items which are 'episodes' that eventually show up in Continue, do not play from the stored position, and similarly, those items when played from the episode grid, do not resume, they start from the beginning. 

Printing the contents of the bookmarks on startup shows correct content ids and saved positions. 


```>>>> DEBUG: value is {"creationDate":"1732138912","position":"184.097"}
>>> DEBUG: key is xyz123
>>>> DEBUG: value is {"creationDate":"1732138983","position":"4.619"}
>>> DEBUG: key is def123
>>>> DEBUG: value is {"creationDate":"1732144234","position":"1.034"}
>>> DEBUG: key is abc123
>>>> DEBUG: value is {"creationDate":"1732138855","position":"126.159"}`

**Comment by rrirower** [comment]
This appears to be a regression bug.

**Comment by rrirower** [comment]
I have a fix for the non-Episodes.  Please try this and let me know if it resolves this part of the issue.  If it does, I'll roll it into a quick update.

In _VideoPlayer.brs_, in the `ShowVideoScreen()` function, please replace:

`m.videoPlayer.addFields( { "playbackId" : playbackId } )`

with:

```
if m.videoPlayer.hasField("playbackId")
     m.videoPlayer.setFields( { "playbackId": playbackId } )
else
     m.videoPlayer.addFields( { "playbackId" : playbackId } )
end if
```

I'm still working on the Episodes part of this issue.

**Comment by Play-AV** [comment]
Will try this in an hour or so and let you know. 

Thanks again :)

**Comment by Play-AV** [comment]
Yeah the change works. Of note, I only have 1 movie item in the channel feed (I can add some dummy items if needed) but the behavior (saving, displaying, and resuming) playback position on movie items appears to be consistent.

**Comment by rrirower** [comment]
Fixed in version 1.0.0.6

**Comment by Play-AV** [comment]
Seems to work, sorry for the late reply :) 
I'll test it more on a larger feed to verify. 
Thanks again.
